docs/advanced_topics/multi_site_multi_instance_multi_tenancy.rst

```diff
@@ -0,0 +1,82 @@
+Multi-site, multi-instance and multi-tenancy
+============================================
+
+This page gives background information on how to run multiple Wagtail sites (with the same source code).
+
+.. contents::
+   :local:
+
+Multi-site
+----------
+
+Multi-site is a Wagtail project configuration where content creators go into a single admin interface and manage the content of multiple websites. Permission to manage specific content, and restricting access to other content, is possible to some extent.
+
+Multi-site configuration is a single code base, on a single server, connecting to a single database. Media is stored in a single media root directory. Content can be shared between sites.
+
+Wagtail supports multi-site out of the box: Wagtail comes with a :class:`site model <wagtail.core.models.Site>`. The site model contains a hostname, port, and root page field. When a URL is requested, the request comes in, the domain name and port are taken from the request object to look up the correct site object. The root page is used as starting point to resolve the URL and serve the correct page.
+
+Wagtail also comes with :ref:`site settings <settings>`. *Site settings* are 'singletons' that let you store additional information on a site. For example, social media settings, a field to upload a logo, or a choice field to select a theme.
+
+Model objects can be linked to a site by placing a foreign key field on the model pointing to the site object. A request object can be used to look up the current site. This way, content belonging to a specific site can be served.
+
+User, groups, and permissions can be configured in such a way that content creators can only manage the pages, images, and documents of a specific site. Wagtail can have multiple *site objects* and multiple *page trees*. Permissions can be linked to a specific page tree or a subsection thereof. Collections are used to categorize images and documents. A collection can be restricted to users who are in a specific group.
+
+Some projects require content editors to have permissions on specific sites and restrict access to other sites. Splitting *all* content per site and guaranteeing that no content 'leaks' is difficult to realize in a multi-site project. If you require full separation of content, then multi-instance might be a better fit...
+
+Multi-instance
+--------------
+
+Multi-instance is a Wagtail project configuration where a single code base is used by multiple websites. Each website has its server process, settings file, and a dedicated database and media directory. This guarantees the total separation of all content.
```

Comment:
each website has its own

Reply:
Also... Is there any particular reason why separate settings files are needed here? I would've thought you could get mostly the same level of control using a single one, with configuration at the env level (which would require no code changes to deploy a new site).

Reply:
The point is, that you can run with multiple settings. Maybe that is possible with environment variables. But adding env vars to the mix doesn't make it clearer what happens. 

Also, env vars are always the same right? How would you differentiate? Maybe 'a single code base' isn't clear enough. I mean, one directory on the server containing the source code, powering all instances. Not a single repo cloned in multiple directories.

Reply:
Okay, maybe I'm misunderstanding this setup in terms of how it actually runs. How do you control which settings file is used by each process?

Reply:
I was imagining some exotic Kubernetes setup with the same docker image powering differently configured nodes.

Reply:
With uWSGI, the acom.ini contains 
`env = DJANGO_SETTINGS_MODULE=settings.acom`
And bcom.ini contains:
`env = DJANGO_SETTINGS_MODULE=settings.bcom`

With a development server, you would run 
`./manage.py 8000 --settings settings.acom`
`./manage.py 8001 --settings settings.bcom`

Using environment variable wound't get you very far. All is executed in the same directory on exactly the same code. Wouldn't the env vars would be similar?

Of course the cost of cloning the repo a second time and have your project code multiple times on disk isn't very expensive. Disk space is cheep. But deploying many instances might be extra work. A single project dir running multiple sites, means a single deploy.

Does that make sense?

Reply:
Yes, that makes sense. I think we could maybe just do with a tweak a here to make that clearer in the text?

Reply:
@ababic I improved this section. Can you have another look?